components/TreasuryAccount/TreasuryPaymentForm.tsx

```diff
@@ -0,0 +1,471 @@
+import React, { useEffect, useState } from 'react'
+import BigNumber from 'bignumber.js'
+import { useRouter } from 'next/router'
+import { AccountInfo } from '@solana/spl-token'
+import { PublicKey } from '@solana/web3.js'
+import { BN } from '@project-serum/anchor'
+import useGovernanceAssets from '@hooks/useGovernanceAssets'
+import useQueryContext from '@hooks/useQueryContext'
+import useRealm from '@hooks/useRealm'
+import { getInstructionDataFromBase64 } from '@models/serialisation'
+import { RpcContext } from '@models/core/api'
+import { Governance } from '@models/accounts'
+import { ParsedAccount } from '@models/core/accounts'
+import { createProposal } from 'actions/createProposal'
+import {
+  getMintMinAmountAsDecimal,
+  getMintNaturalAmountFromDecimal,
+} from '@tools/sdk/units'
+import tokenService from '@utils/services/token'
+import { tryParseKey } from '@tools/validators/pubkey'
+import { debounce } from '@utils/debounce'
+import { notify } from '@utils/notifications'
+import { getTransferInstruction } from '@utils/instructionTools'
+import { precision } from '@utils/formatting'
+import { getTokenTransferSchema } from '@utils/validations'
+import { ProgramAccount, tryGetTokenAccount } from '@utils/tokens'
+import {
+  SendTokenCompactViewForm,
+  UiInstruction,
+} from '@utils/uiTypes/proposalCreationTypes'
+import useTreasuryAccountStore from 'stores/useTreasuryAccountStore'
+import useWalletStore from 'stores/useWalletStore'
+import Button, { SecondaryButton } from '@components/Button'
+import Input from '@components/inputs/Input'
+import TreasuryPaymentIcon from '@components/TreasuryPaymentIcon'
+import { getAccountName } from '@components/instructions/tools'
+import VoteBySwitch from 'pages/dao/[symbol]/proposal/components/VoteBySwitch'
+import { ArrowCircleDownIcon, ArrowCircleUpIcon } from '@heroicons/react/solid'
+
+const TreasuryPaymentForm = ({ close }) => {
+  const currentAccount = useTreasuryAccountStore(
+    (s) => s.compact.currentAccount
+  )
+
+  const connection = useWalletStore((s) => s.connection)
+
+  const {
+    realmInfo,
+    symbol,
+    realm,
+    ownVoterWeight,
+    mint,
+    councilMint,
+    canChooseWhoVote,
+  } = useRealm()
+
+  const { canUseTransferInstruction } = useGovernanceAssets()
+  const { fmtUrlWithCluster } = useQueryContext()
+  const { fetchRealmGovernance } = useWalletStore((s) => s.actions)
+
+  const tokenInfo = useTreasuryAccountStore((s) => s.compact.tokenInfo)
+  const wallet = useWalletStore((s) => s.current)
+  const router = useRouter()
+  const programId: PublicKey | undefined = realmInfo?.programId
+
+  const [form, setForm] = useState<SendTokenCompactViewForm>({
+    destinationAccount: '',
+    amount: 0,
```

Comment:
There shouldn't be default amount, and amount 0 should be accepted in case somebody wants to do dry run proposal